Kernel/FileSystem/Ext2Base.h

```diff
@@ -0,0 +1,431 @@
+    u8  s_reserved_char_pad;
+    u16 s_reserved_pad;  // Padding to next 32bits
+    u32 s_reserved[162]; // Padding to the end of the block
+};
+
+// Codes for operating systems
+enum
+{
+	EXT2_OS_LINUX,
+	EXT2_OS_HURD,
+	EXT2_OBSO_OS_MASIX,
+	EXT2_OS_FREEBSD,
+	EXT2_OS_LITES,
+};
+
+// Revision levels
+enum
+{
+	EXT2_GOOD_OLD_REV, // The good old (original) format
+	EXT2_DYNAMIC_REV,  // V2 format w/ dynamic inode sizes
+};
+
+#define EXT2_CURRENT_REV  EXT2_GOOD_OLD_REV
+#define EXT2_MAX_SUPP_REV EXT2_DYNAMIC_REV
+
+#define EXT2_GOOD_OLD_INODE_SIZE 128
+
+// Journal inode backup types
+#define EXT3_JNL_BACKUP_BLOCKS 1
+
+// Feature set definitions
+#define EXT2_HAS_COMPAT_FEATURE(sb, mask) \
+    (EXT2_SB(sb)->s_feature_compat & (mask))
+#define EXT2_HAS_RO_COMPAT_FEATURE(sb, mask) \
+    (EXT2_SB(sb)->s_feature_ro_compat & (mask))
+#define EXT2_HAS_INCOMPAT_FEATURE(sb, mask) \
+    (EXT2_SB(sb)->s_feature_incompat & (mask))
+
+#define EXT2_FEATURE_COMPAT_DIR_PREALLOC  0x0001
+#define EXT2_FEATURE_COMPAT_IMAGIC_INODES 0x0002
+#define EXT3_FEATURE_COMPAT_HAS_JOURNAL   0x0004
+#define EXT2_FEATURE_COMPAT_EXT_ATTR      0x0008
+#define EXT2_FEATURE_COMPAT_RESIZE_INODE  0x0010
+#define EXT2_FEATURE_COMPAT_DIR_INDEX     0x0020
+#define EXT2_FEATURE_COMPAT_LAZY_BG       0x0040
+
+#define EXT2_FEATURE_RO_COMPAT_SPARSE_SUPER 0x0001
+#define EXT2_FEATURE_RO_COMPAT_LARGE_FILE   0x0002
+/* #define EXT2_FEATURE_RO_COMPAT_BTREE_DIR	0x0004 not used */
+#define EXT4_FEATURE_RO_COMPAT_HUGE_FILE    0x0008
+#define EXT4_FEATURE_RO_COMPAT_GDT_CSUM     0x0010
+#define EXT4_FEATURE_RO_COMPAT_DIR_NLINK    0x0020
+#define EXT4_FEATURE_RO_COMPAT_EXTRA_ISIZE  0x0040
+
+#define EXT2_FEATURE_INCOMPAT_COMPRESSION 0x0001
+#define EXT2_FEATURE_INCOMPAT_FILETYPE    0x0002
+#define EXT3_FEATURE_INCOMPAT_RECOVER     0x0004     // Needs recovery
+#define EXT3_FEATURE_INCOMPAT_JOURNAL_DEV 0x0008 // Journal device
+#define EXT2_FEATURE_INCOMPAT_META_BG     0x0010
+#define EXT3_FEATURE_INCOMPAT_EXTENTS     0x0040
+#define EXT4_FEATURE_INCOMPAT_64BIT       0x0080
+#define EXT4_FEATURE_INCOMPAT_MMP         0x0100
+#define EXT4_FEATURE_INCOMPAT_FLEX_BG     0x0200
+
+#define EXT2_FEATURE_COMPAT_SUPP 0
+#define EXT2_FEATURE_INCOMPAT_SUPP (EXT2_FEATURE_INCOMPAT_FILETYPE)
+#define EXT2_FEATURE_RO_COMPAT_SUPP (EXT2_FEATURE_RO_COMPAT_SPARSE_SUPER | EXT2_FEATURE_RO_COMPAT_LARGE_FILE | EXT4_FEATURE_RO_COMPAT_DIR_NLINK | EXT2_FEATURE_RO_COMPAT_BTREE_DIR)
+
+// Default values for user and/or group using reserved blocks
+#define EXT2_DEF_RESUID 0
+#define EXT2_DEF_RESGID 0
+
+// Default mount options
+#define EXT2_DEFM_DEBUG         0x0001
+#define EXT2_DEFM_BSDGROUPS     0x0002
+#define EXT2_DEFM_XATTR_USER    0x0004
+#define EXT2_DEFM_ACL           0x0008
+#define EXT2_DEFM_UID16         0x0010
+#define EXT3_DEFM_JMODE         0x0060
+#define EXT3_DEFM_JMODE_DATA    0x0020
+#define EXT3_DEFM_JMODE_ORDERED 0x0040
+#define EXT3_DEFM_JMODE_WBACK   0x0060
```

Comment:
Most of those are enums.

Reply:
Thanks for the feedback, but how would you use flags in enums? Would a
```cpp
enum class something
{
    flag0 = (1 << 0),
    flag1 = (1 << 1), // etc
}
```work?

Reply:
Yes that's exactly it, but you can just use the constants from the mount options. If we need binary operators, you can use the AK_BITWISE macro (or whatever the name is) to generate those for you.